Add env vars for the start duration and amount of retries for kube api retries (#1425)
ansible/receptor: docs/source/user_guide/configuration_options.rst

    work-kubernetes:
      - worktype: cat

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Kubernetes Environment Variables
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The following environment variables can be used to configure Kubernetes worker behavior:

.. list-table:: Kubernetes Environment Variables
    :header-rows: 1
    :widths: auto

    * - Variable
      - Description
      - Default value
      - Type

>     * - ``RECEPTOR_KUBE_TIMEOUT_START``

Reviewer: Perhaps its worth noting the valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".

      - Initial timeout duration between Kubernetes API retry attempts (e.g., "1s", "500ms")
      - 1s
      - string (duration)

>     * - ``RECEPTOR_KUBE_RETRY_COUNT``

Reviewer: This is still a fib back off right? 

If yes, then I think these docs need to be re-worded to explain the non linear effect adding a bigger number here will have

    work-kubernetes:
      - worktype: cat

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Kubernetes Environment Variables
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The following environment variables can be used to configure Kubernetes worker behavior:

.. list-table:: Kubernetes Environment Variables
    :header-rows: 1
    :widths: auto

    * - Variable
      - Description
      - Default value
      - Valid range
      - Type
    * - ``RECEPTOR_KUBE_TIMEOUT_START``
      - Initial timeout duration between Kubernetes API retry attempts. Valid time units: "ns", "ms", "s", "m", "h"
      - 1s
      - Any valid duration up to 1m
      - string (duration)
    * - ``RECEPTOR_KUBE_RETRY_COUNT``
      - Number of retry attempts for Kubernetes API operations. Uses exponential backoff with Fibonacci-like sequence.
      - 5
      - 1-100
      - int

**Important Notes:**

> - **Exponential Backoff**: Retry delays increase exponentially using a Fibonacci-like sequence. For example, with ``RECEPTOR_KUBE_TIMEOUT_START=1s``, retry delays will be: 1s, 2s, 3s, 5s, 8s, etc.

Reviewer 2: Genuine question (perhaps a nitpick): is it accurate to describe our backoff as `Exponential` / `exponentially`? Fibonacci is different, right?

Reviewer 3: Just call it "Fibonacci Increment Backoff", or "Fibonacci Backoff"?

> - **Exponential Backoff**: Retry delays increase exponentially using a Fibonacci-like sequence. For example, with ``RECEPTOR_KUBE_TIMEOUT_START=1s``, retry delays will be: 1s, 2s, 3s, 5s, 8s, etc.

Reviewer 3: ```suggestion
- **Fibonacci Backoff**: Retry delays increase by Fibonacci increments. For example, with ``RECEPTOR_KUBE_TIMEOUT_START=1s``, retry delays will be: 1s, 2s, 3s, 5s, 8s, etc.
```

> - **Timeout Start Limit**: ``RECEPTOR_KUBE_TIMEOUT_START`` values exceeding 1 minute will be capped at the maximum of 1 minute.

Reviewer 2: In the code we are removing support for `RECEPTOR_OPEN_LOGSTREAM_TIMEOUT` and reading `RECEPTOR_KUBE_TIMEOUT_START`. Do we need to document that in this guide? Do we have users relying on `RECEPTOR_OPEN_LOGSTREAM_TIMEOUT` that will be confused?

Author: That's a great point! I will document replacing `RECEPTOR_OPEN_LOGSTREAM_TIMEOUT` with the more general `RECEPTOR_KUBE_TIMEOUT_START` which replaces it.
